### PR TITLE
Dev: Add ansible versions to github actions matrix config

### PR DESCRIPTION
**Pull request by philroche**
Dev: Add ansible versions to github actions matrix config

We can then test multiple ansible versions per PR submission

### CODE REVIEW

**Comment by philroche** [comment]
Functional tests currently blocked on devstack install issues which @ccamacho is addressing in https://github.com/os-migrate/os-migrate/pull/492

**Comment by philroche** [comment]
There are branch protection rules that require "build (3.7) " and "functional (3.7) " pass before PR can be merged. In this PR we are however changing the names of the tests as the matrix items are included in the name. 

If we are happy with this change then those branch protection rules need to be changed to include all three build tests and all three functional tests.

**Comment by ccamacho** [comment]
++ on the change in the protection rules, we could after this PR runs, update the protection rules to use the new names, on every run, the protection rule names gets updated, so we could update/merge once it runs.

This change is awesome!

**Comment by ccamacho** [comment]
Just squash this to a single commit and I think is good to go.

File: toolbox/build/venv-requirements-ansible-2.9.txt
Comment (jistr) [inline toolbox/build/venv-requirements-ansible-2.9.txt:1]: All looks good, i'm just wondering how to test the lower-bound supported OpenStack SDK as well. We could add the `openstacksdk` condition here. Or create another job for it in a separate PR?

Overall +1 that we can merge after squashing.

**Comment by ccamacho** [comment]
I suggest to change the commit message like:

Dev: Add ansible versions to github actions matrix config

This commit includes:

- Allowing to test multiple versions of ansible in the
github actions matrix.

- Install required venv packages.

- Updating ansible-lint checks depending on the Ansible version.

**Comment by philroche** [comment]
> I suggest to change the commit message like:
> 
> Dev: Add ansible versions to github actions matrix config
> 
> This commit includes:
> 
> * Allowing to test multiple versions of ansible in the
>   github actions matrix.
> * Install required venv packages.
> * Updating ansible-lint checks depending on the Ansible version.

Thanks. Now squashed

File: toolbox/build/venv-requirements-ansible-2.9.txt
Comment (philroche) [reply toolbox/build/venv-requirements-ansible-2.9.txt:1]: I can add a new PR for that

**Review by jistr — APPROVED** [review]

**Pull request merged**
